packages/contracts-rfq/contracts/FastBridgeV2.sol

```diff
@@ -77,18 +77,20 @@ contract FastBridgeV2 is Admin, IFastBridgeV2, IFastBridgeV2Errors {
 
     /// @inheritdoc IFastBridge
     function dispute(bytes32 transactionId) external onlyRole(GUARD_ROLE) {
-        if (bridgeTxDetails[transactionId].status != BridgeStatus.RELAYER_PROVED) revert StatusIncorrect();
-        if (_timeSince(bridgeTxDetails[transactionId].proofBlockTimestamp) > DISPUTE_PERIOD) {
+        BridgeTxDetails storage $ = bridgeTxDetails[transactionId];
+
+        if ($.status != BridgeStatus.RELAYER_PROVED) revert StatusIncorrect();
+        if (_timeSince($.proofBlockTimestamp) > DISPUTE_PERIOD) {
             revert DisputePeriodPassed();
         }
 
-        address disputedRelayer = bridgeTxDetails[transactionId].proofRelayer;
+        address disputedRelayer = $.proofRelayer;
```

Comment:
One extra change that allows the compiler optimizer to do its job. If you do multiple storage reads in a row (without any instructions in-between), I believe it is able to actually read it just once, if all the variables reside in the same storage slot (and they do, since it's a packed struct to fit within a single storage slot).

You can check this by running a gas bench against the optimized suggestion, and then moving the `disputedRelayer` assignment back after the checks - you should notice a slight gas increase.

```suggestion
        BridgeStatus status = $.status;
        uint40 proofBlockTimestamp = $.proofBlockTimestamp;
        address disputedRelayer = $.proofRelayer;
        if (status != BridgeStatus.RELAYER_PROVED) revert StatusIncorrect();
        if (_timeSince(proofBlockTimestamp) > DISPUTE_PERIOD) {
            revert DisputePeriodPassed();
        }
```